crates/bevy_reflect/src/serde/de.rs

```diff
@@ -378,6 +400,11 @@ impl<'a, 'de> DeserializeSeed<'de> for TypedReflectDeserializer<'a> {
             return Ok(value);
         }
 
```

Comment:
I don't have a good feeling of whether this new block should be after or before the one above. Shouldn't we first try to deserialize with the help of the registry, if that allows customized deserializing, and fallback to the one without if not (so, the reverse order)?

Reply:
(and if there's an argument for the current order, can you please document it in code?)

Reply:
Yeah I think that's a good idea.

I wonder if we should just merge `ReflectDeserialize` with this one, since `DeserializeReflect` has a blanket implementation for any `T: Reflect + Deserialize`. 🤔

Reply:
If we can that would certainly remove a chunk of confusion for users.

Reply:
I think we can do this in a followup PR just to keep the discussion a bit more focused. We can also consider [removing the blanket impl](https://github.com/bevyengine/bevy/pull/8611#discussion_r1536867429) until then as well.

Reply:
Yes let's merge this and iterate if needed with a separate change.